### PR TITLE
Nrf security v1.0.0 fixes

### PR DESCRIPTION
**Pull request by tejlmand**
Several fixes for nrf_security found during testing.

Fix 1:
Files:
    - pk_wrap.c
    - cipher_wrap.c
    - md_wrap.c
are now placed in an independent library so that
the correct implementation from either glue layer, or the active
backend is always used.

Fix 2:    
 The nrf_security Kconfig for OFB, CFB, XTS, and GCM were not specifying
VANILLA_MBEDTLS_x for those symbols when selecting vanilla backend.
VANILLA_MBEDTLS_x defines has now been specified to ensure the
corresponding symbols are not renamed while still allow for future extension.

Fix 3:   
    When compiling mbedtls_vanilla then not all Kconfig settings where
    converted into mbedtls corresponding defines when compiling
    mbedtls_vanilla.
    
    This caused certain ciphers suites not be availble for TLS sessions
    although they were part of the mbedcrypto library.

### CODE REVIEW

File: nrf_security/src/mbedtls/CMakeLists.txt [hunk 1]
```diff
@@ -133,11 +131,17 @@ if(TARGET ${IMAGE}mbedcrypto_cc310)
...
+# In order to ensure correct linking to glue functions from the following files
+# those must be placed in independent library to ensure they will always link
+# to the correct renamed symbols or glue layer.
+# In no symbols has been renamed they will link to correct implementation in
+# selected backend library.
+zephyr_library_named(mbedwrap)
 zephyr_library_sources(${ARM_MBEDTLS_ROOT}/library/ctr_drbg.c)
+zephyr_library_sources(${ARM_MBEDTLS_ROOT}/library/pk_wrap.c)
+zephyr_library_sources(${ARM_MBEDTLS_ROOT}/library/md_wrap.c)
+zephyr_library_sources(${NRF_MBEDTLS_ROOT}/src/mbedtls/replacements/cipher_wrap.c)
```
Comment (frkv) [inline nrf_security/src/mbedtls/CMakeLists.txt]: Addition of CONFIG_MBEDTLS_CIPHER_AES_256_ECB_C, CONFIG_MBEDTLS_CIPHER_AES_256_CBC_C, CONFIG_MBEDTLS_CIPHER_AES_256_CTR_C and CONFIG_MBEDTLS_CIPHER_AES_256_CCM_C here as well?

Comment (tejlmand) [reply nrf_security/src/mbedtls/CMakeLists.txt]: yep, fixed

**Comment by thst-nordic** [comment]
this repo obviously doesn't have any CI testing yet, but this change has passed in the nRF repo.

**Comment by thst-nordic** [comment]
CI: doc error 

tests.test_check_warnings: DEBUG: /jenkins_workspace/workspace/connect-nrf_documentation_master/_build/nrfxlib/rst/nrf_security/README.rst:11: WARNING: toctree contains reference to nonexisting document 'nrf_security/CHANGELOG'

File: nrf_security/CHANGELOG.rst [hunk 2]
```diff
@@ -1,12 +0,0 @@
-.. _changelog_nrf_security:
```
Comment (frkv) [inline nrf_security/CHANGELOG.rst]: This erased CHANGELOG.rst is still referenced in the toctree in nrf_security

Comment (frkv) [reply nrf_security/CHANGELOG.rst]: I have amended the commit to remove this toctree reference, but this doesn't start the automatic build of this nor the referencing Prf #882. @thst-nordic Any Idea why?

Comment (tejlmand) [reply nrf_security/CHANGELOG.rst]: Cause that is not picked up by https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/882
I have re-triggered the build now.

Comment (frkv) [reply nrf_security/CHANGELOG.rst]: Great, marking as resolved

**Comment by tejlmand** [comment]
Obsolete

**Pull request closed without merging**